lib/Runtime/Library/JavascriptRegularExpression.cpp

```diff
@@ -671,11 +671,37 @@ namespace Js
         Assert(!(callInfo.Flags & CallFlags_New));
         Assert(args.Info.Count > 0);
 
-        JavascriptRegExp* obj = GetJavascriptRegExp(args, _u("RegExp.prototype.toString"), scriptContext);
+        PCWSTR const varName = _u("RegExp.prototype.toString");
 
-        bool sourceOnly = false;
-        bool useFlagsProperty = scriptContext->GetConfig()->IsES6RegExPrototypePropertiesEnabled();
-        return obj->ToString(sourceOnly, useFlagsProperty);
+        const ScriptConfiguration* scriptConfig = scriptContext->GetConfig();
+
+        if (scriptConfig->IsES6RegExPrototypePropertiesEnabled())
+        {
+            RecyclableObject *thisObj = GetThisObject(args, varName, scriptContext);
+            JavascriptString* source = JavascriptConversion::ToString(
+                JavascriptOperators::GetProperty(thisObj, PropertyIds::source, scriptContext),
+                scriptContext);
+            JavascriptString* flags = JavascriptConversion::ToString(
+                JavascriptOperators::GetProperty(thisObj, PropertyIds::flags, scriptContext),
+                scriptContext);
+
+            CharCount length = source->GetLength() + flags->GetLength() + 2; // 2 for the two '/'s
+            CompoundString *const builder =
+                CompoundString::NewWithCharCapacity(length, scriptContext->GetLibrary());
+            builder->Append(_u('/'));
+            builder->Append(source);
+            builder->Append(_u('/'));
+            builder->Append(flags);
+            return builder;
+        }
+        else
+        {
+            JavascriptRegExp* obj = GetJavascriptRegExp(args, varName, scriptContext);
+
+            bool sourceOnly = false;
+            bool useFlagsProperty = scriptConfig->IsES6RegExPrototypePropertiesEnabled();
+            return obj->ToString(sourceOnly, useFlagsProperty);
```

Comment:
Is this the only place the ::ToString gets called? wondering if we can remove useFlagProperty altogether. In the ToString itself can check of IsES6RegExPrototypePropertiesEnabled() and do the work.